cfp/tests/test_views.py

```diff
@@ -119,6 +119,52 @@ def test_post_when_not_authorized(self):
         self.assertContains(rsp, 'Only the proposer of a proposal can update the proposal')
 
 
+class ProposalDeleteTests(TestCase):
+    @classmethod
+    def setUpTestData(cls):
+        cls.alice = factories.create_user('Alice')
+        cls.bob = factories.create_user('Bob')
+
+    def test_get(self):
+        self.client.force_login(self.alice)
+        proposal = factories.create_proposal(self.alice)
+        rsp = self.client.get(f'/cfp/proposals/{proposal.proposal_id}/delete/')
+        self.assertEqual(rsp.status_code, 405)
+        self.assertEqual(Proposal.objects.get(id=proposal.id), proposal)
+
+    def test_post(self):
+        self.client.force_login(self.alice)
+        proposal = factories.create_proposal(self.alice)
+        rsp = self.client.post(f'/cfp/proposals/{proposal.proposal_id}/delete/', follow=True)
+        self.assertContains(rsp, 'Your proposal has been withdrawn', html=True)
+
+        with self.assertRaises(Proposal.DoesNotExist):
+            Proposal.objects.get(id=proposal.id)
```

Comment:
FYI, I think you could use `self.assertFalse(Proposal.objects.filter(id=proposal.id).exists())`.